### PR TITLE
All about Headers

### PR DESCRIPTION
**Pull request by tjvr**
Here are some things I'm working on for you @technoboy10 

[relevant SO answer](http://stackoverflow.com/questions/10369679/do-http-proxy-servers-modify-request-packets#answer-10370170) for some of this stuff

TODO: [find a large file on a host with CORS disabled](https://github.com/technoboy10/crossorigin.me/pull/68/commits/0e9e3b28b6d8aba95320a3d628f493846efac4f7#diff-3896ea03d1c798cd42263c5ff63e97f8R6)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,10 @@
   "description": "A CORS proxy for everyone",
   "main": "index.js",
   "scripts": {
-    "test": "make test"
+    "test": "npm run lint && npm run unit",
```
Comment (technoboy10) [inline package.json]: Aren't these changes a duplicate of #67?

File: tests/integration/cors.js [hunk 2]
```diff
@@ -6,9 +6,16 @@ describe('Simple CORS request', function () {
     it('should include the correct headers', function (done) {
         server
         .get('/https://google.com')
+        .set('Origin', 'http://example.com')
```
Comment (technoboy10) [inline tests/integration/cors.js]: Requiring origin header should be in a separate PR as mentioned above (also, would error 412 be better here?)

File: package.json (hunk 1 above)
Comment (tjvr) [inline package.json]: yes this is based on that PR you see!

Comment (tjvr) [inline package.json]: if you merge the other one, i will rebase this one onto v2, which hopefully should make github happy

File: tests/integration/limits.js [hunk 3]
```diff
@@ -3,12 +3,12 @@ const supertest = require('supertest');
...
 describe('Large file request', function () {
-    it('should only serve 2MB of data', function (done) { // Will attempt to fetch a 4.3MB file
-        server
-        .get('/https://upload.wikimedia.org/wikipedia/commons/b/bc/Boulder_Colorado_-_Perl_Street_Mall_-2005-10-14T204331.png') //eslint-disable-line
-        .expect(function (res) {
-            res.body.length < 2e6;
-        })
-        .expect(200, done);
-    });
+    // TODO find a large file on a host with CORS disabled
```
Comment (tjvr) [inline tests/integration/limits.js]: because that test is broken!

File: tests/integration/cors.js (hunk 2 above)
Comment (tjvr) [inline tests/integration/cors.js]: Whatever makes you happy :)

Regarding 412: not sure?

File: modules/proxy.js [hunk 4]
```diff
@@ -1,16 +1,69 @@
...
+
 /*
 get handler handles standard GET reqs as well as streams
 */
 function get (req, res, next) {
     res.header('Access-Control-Allow-Origin', '*'); // Actually do the CORS thing! :)
 
+    var url = req.params[0];
+
+    // require CORS header
```
Comment (tjvr) [inline modules/proxy.js]: ok!

File: tests/integration/limits.js (hunk 3 above)
Comment (technoboy10) [inline tests/integration/limits.js]: It seemed to work OK for me, why do you say it's broken?

Comment (technoboy10) [inline tests/integration/limits.js]: Ohhhhhh I see cors is enabled on Wikimedia? I can host a non-CORS photo somewhere else if that heps

Comment (technoboy10) [inline tests/integration/limits.js]: Can you re-enable the test with the image https://01.keybase.pub/big.jpg?

**Comment by tjvr** [comment]
no u reenable it

**Comment by technoboy10** [comment]
k :P
